src/libs/auth.ts

```diff
@@ -4,22 +4,43 @@
 
 import { HTTP } from '../http';
 import { MyMICDSOptions } from '../options';
-import { Observable } from 'rxjs/Observable'; // tslint:disable-line
+
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { Observable } from 'rxjs/Observable';
 import { tap } from 'rxjs/operators';
 
+import { decode } from 'jsonwebtoken';
+
 export class AuthAPI {
 
-	constructor(private http: HTTP, private options: MyMICDSOptions) { }
+	auth$: BehaviorSubject<AuthSnapshot>;
+	authSnapshot: AuthSnapshot;
+
+	constructor(private http: HTTP, private options: MyMICDSOptions) {
+		this.authSnapshot = { jwt: undefined };
+		this.auth$ = new BehaviorSubject(this.authSnapshot);
+	}
 
 	login(param: LoginParameters) {
 		return this.http.post<LoginResponse>('/auth/login', param).pipe(
-			tap(r => this.options.jwtSetter(r.jwt, param.remember))
+			tap(r => {
+				this.options.jwtSetter(r.jwt, param.remember);
+				const parsed = decode(r.jwt);
+				if (parsed && typeof parsed !== 'string') {
+					this.authSnapshot = {jwt: parsed as JWT};
+					this.auth$.next(this.authSnapshot);
+				}
+			})
 		);
 	}
 
 	logout() {
 		return this.http.post('/auth/logout').pipe(
-			tap(() => this.options.jwtClear())
+			tap(() => {
+				this.options.jwtClear();
+				this.authSnapshot = { jwt: undefined };
+				this.auth$.next(this.authSnapshot);
```

Comment:
In the Angular front-end, `undefined` is when the user's auth state hasn't been determined yet, and `null` is when auth state has been determined but just is logged out

Reply:
Thats a really good way to distinguish, but is there any practical use? The login state being unknown should be the same as being logged out in the frontend right?

Reply:
I cannot do `new BehaviorSubject(undefined)` because typescript wants the default value to be the type `AuthSnapshot`

Reply:
I guess since the JWT getter is treated synchronously we don't need any "pending" value anyways. But we still want to have it initially check if already logged in